src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorAbsSendTime.java

```diff
@@ -0,0 +1,719 @@
+                break;
+            }
+        }
+        return bestIt;
+    }
+
+    @Override
+    public void run(){}
+
+    ProbeResult processClusters(long nowMs)
+    {
+        synchronized (critSect) {
+            List<Cluster> clusters = new ArrayList<Cluster>();
+            computeClusters(clusters);
+            if (clusters.isEmpty()) {
+                // If we reach the max number of probe packets and still
+                // have no clusters,
+                // we will remove the oldest one.
+                if (probes_.size() >= kMaxProbePackets)
+                    probes_.remove(0);
+                return ProbeResult.kNoUpdate;
+            }
+
+            Cluster bestProbe = findBestProbe(clusters);
+            int probeBitrateBps =
+                    Math.min(bestProbe.getSendBitrateBps(), bestProbe
+                            .getRecvBitrateBps());
+            // Make sure that a probe sent on a lower bitrate
+            // than our estimate can't
+            // reduce the estimate.
+            if (isBitrateImproving(probeBitrateBps))
+            {
+                logger.warn("Probe successful, sent at "
+                        + bestProbe.getSendBitrateBps() +
+                        " bps, received at "
+                        + bestProbe.getRecvBitrateBps()
+                        + " bps. Mean send delta: " + bestProbe.sendMeanMs
+                        + " ms, mean recv delta: " + bestProbe.recvMeanMs
+                        + " ms, num probes: " + bestProbe.count);
+                remoteRate.setEstimate(probeBitrateBps, nowMs);
+                return ProbeResult.kBitrateUpdated;
+            }
+            // Not probing and received non-probe packet,
+            // or finished with current set  of probes.
+            if (clusters.size() >= kExpectedNumberOfProbes)
+                probes_.clear();
+            return ProbeResult.kNoUpdate;
+        }
+    }
+
+    boolean isBitrateImproving(int newBitrateBps)
+    {
+        synchronized (critSect)
+        {
+            boolean initialProbe = !remoteRate.isValidEstimate()
+                    && newBitrateBps > 0;
+            boolean bitrateAboveEstimate =
+                    remoteRate.isValidEstimate() &&
+                            newBitrateBps > (int) (remoteRate
+                                    .getLatestEstimate());
+            return initialProbe || bitrateAboveEstimate;
+        }
+    }
+
+    /**
+     * Reverse-transforms a specific packet.
+     *
+     * @param packet the transformed packet to be restored.
+     * @return the restored packet.
+     */
+    @Override
+    public RawPacket reverseTransform(
+            RawPacket packet)
+    {
+        if (absoluteSendTimeEngine
+                .hasAbsoluteSendTimeExtension(packet) == null) {
+            logger.warn( "RemoteBitrateEstimatorAbsSendTimeImpl: Incoming" +
+                    " packet is missing absolute send time extension!");
+        return null;
```

Comment:
This drops the packet. This shouldn't happen just because it doesn't have the extension. I'd expect the wrapper to handle this logic.